OSDOCS-17626 updated zero-trust release notes for GA (#103679)
openshift/openshift-docs: security/zero_trust_workload_identity_manager/zero-trust-manager-release-notes.adoc

{zero-trust-full} supports the following components and versions:

[cols="1,1",options="header"]
|===
| Component
| Version

| SPIRE Server
| 1.13.3

| SPIRE Agent
| 1.13.3

| SPIRE Controller Manager
| 0.6.3

| SPIRE OIDC Discovery Provider
| 1.13.3

| SPIFFE CSI Driver
| 0.2.8
|===

[id="zero-trust-manager-1-0-0-features-enhancements_{context}"]
=== New features and enhancements

SPIRE federation support::

The Operator now includes support for SPIRE federation, enabling workloads across distinct trust domains to securely communicate and authenticate with each other.

* Key capabilities:

** Configuration of bundle endpoints using `https_spiffe` (mTLS) or `https_web` (Web PKI) profiles.
** Automatic certificate management via the ACME protocol (e.g., Let's Encrypt).
** Automatic {product-title} route creation for federation endpoints.
** Ability to configure relationships with multiple federated trust domains.

> * Customer Action Required:

Author: Done

* Verification types:

** `auto` (default): Verification utilizes OpenShift defaults (`/etc/kubernetes/kubelet-ca.crt`).
** `hostCert`: Uses a custom CA certificate path.
** `skip`: Skips TLS verification (not recommended for production use).

Configurable Certificate Authority and JSON Web Token key types::

Administrators can now configure the cryptographic key types used for the SPIRE Server Certificate Authority (CA) and JSON Web Token (JWT) signing, ensuring compliance with organizational security policies.

* Supported Key Types: `rsa-2048` (default), `rsa-4096`, `ec-p256`, `ec-p384`.

* Customer action required:

** Review organizational security policies to determine required key types.

Custom namespace deployment::

* The Operator and all associated operands can now be deployed within a custom namespace, providing flexibility for organizations with specific namespace governance requirements.

Proxy-aware Operator and operands::

* The Operator and all managed operands are now proxy-aware and automatically inherit cluster-wide proxy settings when configured.

Enhanced Security Context Constraints::

> * SPIRE Agent and SPIFFE CSI Driver now operate under the restricted Security Context Constraints (SCC).

Reviewer: ```suggestion
* SPIRE server and OIDC Discovery Provider now operate under the restricted Security Context Constraints (SCC).
```

* The Operator and all operand containers are configured with the `ReadOnlyRootFilesystem` set to `true`.

Enhanced API validation::

Comprehensive Common Expression Language (CEL) validation has been integrated into all Custom Resource Definitions (CRDs) to prevent configuration errors during admission control.

* Key validations:

** All Operator CRDs are enforced as singletons (must be named `cluster`).

> ** Immutable Fields: Fields including `trustDomain`, `clusterName`, `bundleConfigMap`, federation `bundleEndpoint`, and all `Persistence` settings (`size`, `accessMode`, and `storageClass`) are now immutable after initial creation.

Author: Done

* Operator: The operator’s log verbosity is now configurable via the `OPERATOR_LOG_LEVEL` environment variable using klog’s `textlogger`.

Refactor for create-only mode::

By setting the `CREATE_ONLY_MODE` environment variable, users can prevent the operator from reconciling updates. This allows for manual resource modification without interference. If this mode is disabled, the Operator resumes enforcing the state and overwrites any manual changes.

=== Status and observability improvements

Enhanced status reporting::

* The main CR now aggregates status information from all operand CRs.

* New status conditions include Upgradeable (indicating a safe upgrade path) and Progressing (detailing deployment progress).

Operator metrics::

* Operator metrics are now exposed and secured with appropriate RBAC configuration.

* Integration is supported with the OpenShift monitoring stack.

[id="zero-trust-manager-1-0-0-bug-fixes_{context}"]
=== Fixed issues

Enhanced Security Context Constraints for SPIRE Agent::
+

> Before this update, the SPIRE Agent container had unrestricted capabilities and host network access, leading to a potential container security violation and potential privilege escalation. With this release, Security Context Constraints (SCC) for SPIRE Agent and SPIFFE CSI Driver have been restricted, ensuring the Spire agent runs securely with only necessary capabilities, reducing potential security risks for the end user.

Author: Done

+
(link:https://issues.redhat.com/browse/SPIRE-60[SPIRE-60])

SpireServer updates now propagate without operator restart::
+
* Before this update, the operator failed to trigger reconciliation after updating the operand CR spec. As a consequence, user updates to SpireServer CR resources were not propagated to the StatefulSet, causing reconciliation to fail and changes to be ignored, leading to inconsistent resource allocation. With this release, we've fixed the race between the manager and reconciler's cache to trigger reconciliation after CR updates. As a result, day2 patch operations on SpireServer CRs will reliably trigger reconciliation, ensuring updated values are applied to the StatefulSet without manual operator restart.
+
(link:https://issues.redhat.com/browse/SPIRE-68[SPIRE-68])

Removed unnecessary security context constraint for OpenID Connect discovery provider::
+

> * Before this update, the system unnecessarily created a custom security context constgraint (SCC) for the OpenID Connect (OIDC) discovery provider, which increased the security footprint and configuration complexity even though the deployment did not require it. With this release, the custom SCC creation logic has been removed, resulting in a configuration where the OIDC discovery provider operates successfully without the extra security constraints.

Reviewer: ```suggestion
* Before this update, the system unnecessarily created a custom security context constraint (SCC) for the OpenID Connect (OIDC) discovery provider, which increased the security footprint and configuration complexity even though the deployment did not require it. With this release, the custom SCC creation logic has been removed, resulting in a configuration where the OIDC discovery provider operates successfully without the extra security constraints.
```

Typo

Author: Done

+
(link:https://issues.redhat.com/browse/SPIRE-190[SPIRE-190])

> Fixed Config Map Reconciliation for SPIRE Controller Manager::

Reviewer: ```suggestion
Fixed ConfigMap Reconciliation for SPIRE Controller Manager::
```
Nits: consistent casing

+

> Before this update, Spire-controller manager config map reconciliation failed due to an unhandled edge case in the previous implementation. As a consequence, users experienced configuration inconsistencies. With this release, the Spire-controller manager config map reconciliation issue has been resolved. As a result, end users now experience seamless Spire-controller manager configuration.

Author: Done

toc::[]

> 

Reviewer 2: This file needs to have CQA 2.0 compliance. Abstract etc.

Author: Yeah, I know the headings are wrong. We will eventually put them into "modules" but that's for another time.

These release notes track the development of {zero-trust-full}.

[id="zero-trust-manager-release-notes-1-0-0"]
== {zero-trust-full} 1.0.0 (General Availability)

Issued: 2025-12-17

This release of the {zero-trust-full} introduces new capabilities focused on enterprise readiness, security, and operational flexibility. Key features include SPIRE federation support for cross-cluster identity, PostgreSQL database support for production persistence, and enhanced security controls implemented through stricter Security Context Constraints (SCCs) and comprehensive API validation.

The following advisories are available for the {zero-trust-full}.

* https://access.redhat.com/errata/RHBA-2025:23438[RHBA-2025:23438]
* https://access.redhat.com/errata/RHBA-2025:23439[RHBA-2025:23439]
* https://access.redhat.com/errata/RHBA-2025:23440[RHBA-2025:23440]
* https://access.redhat.com/errata/RHBA-2025:23441[RHBA-2025:23441]
* https://access.redhat.com/errata/RHBA-2025:23442[RHBA-2025:23442]
* https://access.redhat.com/errata/RHBA-2025:23443[RHBA-2025:23443]
* https://access.redhat.com/errata/RHBA-2025:23446[RHBA-2025:23446]

> This release of {zero-trust-full} is a Technology Preview.

Reviewer 2: You can use the tech preview snippet here.

Author: Done

> 

Author: Done

* Key capabilities:

** Configuration of bundle endpoints using `https_spiffe` (mTLS) or `https_web` (Web PKI) profiles.
** Automatic certificate management via the ACME protocol (e.g., Let's Encrypt).
** Automatic {product-title} route creation for federation endpoints.
** Ability to configure relationships with multiple federated trust domains.

* Customer Action Required:

** Review the `federation` configuration within the `SpireServer Custom Resource (CR).
** Ensure proper DNS resolution and network connectivity to federated trust domains.

[id="zero-trust-manager-1-0-0-configurable-socketpath-plugin-name_{context}"]
==== Configurable agent socket path and Container Storage Interface (CSI) plugin name

The SPIRE Agent socket path and the SPIFFE CSI Driver plugin name are now configurable, providing operational flexibility for environments with specific directory requirements or co-existence with multiple SPIFFE deployments.

* Key configuration points:

** `SpireAgent.spec.socketPath`
** `SpiffeCSIDriver.spec.agentSocketPath`

> ** `ApiffeCSIDriver.spec.pluginName`

Author: Done

* Customer action required:

** Ensure consistency between `socketPath` in the `SpireAgent` CR and `agentSocketPath` in the `SpiffeCSIDriver` CR.

[id="zero-trust-manager-1-0-0-workload-attestors_{context}"]
==== Workload attestors verification API

A new API has been introduced to configure kubelet certificate verification for workload attestation, enhancing security and supporting various OpenShift configurations.

* Verification types:

** `auto` (default): Verification utilizes OpenShift defaults (`/etc/kubernetes/kubelet-ca.crt`).
** `hostCert': Uses a custom CA certificate path.

> ** `skip``: Skips TLS verification (not recommended for production use).

Author: Done

* Customer action required:

** Review existing CR configurations to ensure compliance with the new validation rules.

[id="zero-trust-manager-1-0-0-common-configuration_{context}"]
==== Common configuration consolidation

* Standard configuration options (`labels`, `resources`, `affinity`, `tolerations`, `nodeSelector`) are now standardized across all operand CRs via a shared `CommonConfig` structure.

[id="zero-trust-manager-1-0-0-immutable-fields_{context}"]
==== Immutable fields

The following fields are now strictly enforced as *immutable* after initial creation:

[cols="1,1,1",options="header"]
|===
| Custom resource
| Field
| Notes

| ZeroTrustWorkloadIdentityManager
| `trustDomain`, `clusterName`, `bundleConfigMap`
| Cannot be modified post-creation.

| SpireServer
| `persistence.size`, `persistence.accessMode`, `persistence.storageClass`
| Cannot be modified post-creation.

| SpireServer
| `federation.bundleEndpoint`
| Cannot be removed once configured.
|===

> 

Reviewer 3: consider removing this table as they has been mentioned at this line already: https://github.com/openshift/openshift-docs/pull/103679/files#diff-cae3550cca1c8918c4da09faef482503a73e4be67ba89e4e2b5259044e9de140R138

[id="zero-trust-manager-1-0-0-status-observability-improvements_{context}"]
=== Status and observability improvements

[id="zero-trust-manager-1-0-0-enhanced-status-reporting_{context}"]
==== Enhanced status reporting

* The main CR now aggregates status information from all operand CRs.

* New status conditions include Upgradeable (indicating a safe upgrade path) and Progressing (detailing deployment progress).

[id="zero-trust-manager-1-0-0-operator-metrics_{context}"]
==== Operator metrics

* Operator metrics are now exposed and secured with appropriate RBAC configuration.

* Integration is supported with the OpenShift monitoring stack.

[id="zero-trust-manager-1-0-0-bug-fixes_{context}"]

> === Bug fixes

Author: Done
